### PR TITLE
Fixed Timezone issue and went defensive

### PR DESCRIPTION
**Pull request by cloudnautique**
Addresses issue #74 by adding defensive try/catch blocks. Also,
addressed the handling of timezones that caused issue part1.

While in here, added code to address issue #73. Stats are only added
when the statas are requested in the ping. Otherwise, they will be none.

### CODE REVIEW

File: cattle/plugins/docker/compute.py [hunk 1]
```diff
@@ -81,6 +82,16 @@ def _add_resources(self, ping, pong):
         if not utils.ping_include_resources(ping):
             return
```
Comment (ibuildthecloud) [inline cattle/plugins/docker/compute.py]: sorry, I hate to comment on small things like this.  But there would be less code if you did
```python
stats = None
try:
    stats = ....
except:
   log.exception(...)
```
You default the stats to None and only set it on success

File: cattle/plugins/host_info/main.py [hunk 2]
```diff
@@ -14,6 +14,9 @@ def __init__(self):
     def collect_data(self):
         data = {}
         for collector in self.collectors:
-            data[collector.key_name()] = collector.get_data()
+            try:
+                data[collector.key_name()] = collector.get_data()
+            except:
```
Comment (ibuildthecloud) [inline cattle/plugins/host_info/main.py]: Do you want to `log.exception(...)` here so we know whats up?

File: cattle/utils.py [hunk 3]
```diff
@@ -100,9 +100,10 @@ def timestamp_diff(self, time_current, time_prev):
         return round((diff * 10**9))
 
     def _timestamp_convert(self, stime):
-        # This is a bit hacky
-        t_strip = stime[:-4]
-        return datetime.strptime(t_strip, "%Y-%m-%dT%H:%M:%S.%f")
+        # Cadvisor handles everything in nanoseconds.
+        # Python does not.
+        t_conv = arrow.get(stime[0:26])
```
Comment (ibuildthecloud) [inline cattle/utils.py]: even with arrow you have to trim the string?

File: cattle/plugins/docker/compute.py [hunk 4]
```diff
@@ -81,6 +82,16 @@ def _add_resources(self, ping, pong):
...
 
+        if utils.ping_include_stats(ping):
+            try:
+                stats = self.host_info.collect_data()
+            except:
+                error = sys.exc_info()
```
Comment (ibuildthecloud) [inline cattle/plugins/docker/compute.py]: I though log.exception did this magically.  I could be wrong.

File: cattle/utils.py (hunk 3 above)
Comment (cloudnautique) [inline cattle/utils.py]: Yeah, Python can't handle nanoseconds. Arrow is much more graceful dealing with the strings.

File: cattle/plugins/docker/compute.py (hunk 4 above)
Comment (cloudnautique) [inline cattle/plugins/docker/compute.py]: Another check of the docs clarified, it does handle magic. Will remove.

**Comment by cloudnautique** [comment]
Updated, but needs squashing before merge. @ibuildthecloud @cjellick

**Pull request merged**